app_test.go

```diff
@@ -3077,3 +3077,64 @@ func TestFlagAction(t *testing.T) {
 		})
 	}
 }
+
+func TestPersistentFlag(t *testing.T) {
+
+	var vflag1, vflag2, vflag3 int
+
+	a := &App{
+		Commands: []*Command{
+			{
+				Name: "foo",
+				Flags: []Flag{
+					&IntFlag{
+						Name:        "flag1",
+						Destination: &vflag1,
+					},
+					&IntFlag{
+						Name:        "flag2",
+						Persistent:  true,
+						Destination: &vflag2,
+					},
+				},
+				Subcommands: []*Command{
+					{
+						Name: "bar",
```

Comment:
I would greatly appreciate more descriptive flag and command names here, both for human comprehension and greppability 😁 Something along the lines of `var topInt, topPersistentInt, subCommandInt int` and commands named `"top-test-persistent-flag"` and `"sub-test-persistent-flag"`?

Reply:
Yes nice idea. Updated